examples/kernel-dynamic-tile.cpp

```diff
@@ -0,0 +1,29 @@
+#include "RAJA/RAJA.hpp"
+
+int main(int argc, char* argv[])
+{
+
+//Using policy = KernelPolicy<Tile<tile_dynamic<0>, seq_exec, …>>;
+//RAJA::kernel_param<policy>(
+// make_tuple(RangeSegment(0,N)),
+//  make_tuple(32),  // param 0 is referenced by tile_dynamic
+//  [=](int i, int tile_size){
+//
+//  });
+
+  using namespace RAJA;
+
+  kernel_param<
+    KernelPolicy< 
+      statement::Tile<0, statement::tile_dynamic<0>, seq_exec,
```

Comment:
Would it make sense for us to wrap ParameterId's and ArgumentId's in types?
Like
statement::Tile<Arg<0>, statement::tile_dynamic<Param<0>>, seq_exec, ...